### ISSUE TITLE
Highlight breaks when *ngFor *ngIf is in the element tag

### ISSUE THREAD

**Issue by karkianish** [vincaslt/vscode-highlight-matching-tag#21]
The extension is unable to find the matching pair, if any of the child element has *ngIf/*ngFor on the element tag. '*' may be the culprit.

VSCode version: 1.18.0

```
              <div> <!-- cant find -->
                    <div> <!-- cant find -->
                        <div class="some-class" *ngFor="let bar of bars"> <!-- cant find -->
                            <div class="foo"> <!--works!-->
                                <p>Hello</p> <!--works-->
                            </div>
                            <div class="bar"><!--works!-->
                                <p>Hello</p> <!--works-->
                            </div>
                        </div>
                    </div>
                </div>
```

**Comment by adarshmadrecha** [comment]
The plugin working fine
![hilighting1](https://user-images.githubusercontent.com/11911938/33118300-612b6730-cf91-11e7-8516-8fb3194a2975.JPG)

Plugin not working
![hilighting2](https://user-images.githubusercontent.com/11911938/33118298-60dcd61a-cf91-11e7-8ca4-a3cf433d1410.JPG)

**Comment by vincaslt** [comment]
Should be fixed with v0.5, test cases cover it now.
